angular/projects/admin-nrpti/src/app/records/inspections/inspection-lng-detail/inspection-lng-detail.component.ts

```diff
@@ -114,7 +114,13 @@ export class InspectionLNGDetailComponent extends RecordComponent implements OnI
       return 'Not Published';
     }
 
-    return `Published on ${this.datePipe.transform(this.data.datePublished, 'mediumDate')}`;
+    const publishDate = this.datePipe.transform(this.data && this.data.datePublished, 'mediumDate');
```

Comment:
I'm not sure why we're doing logic here, instead of pushing it into a pipe in the template.

Reply:
Just because the logic in this function is probably a bit complex to put all inline in the template.

It does checks on this.data && this.data.datePublished to ensure its not null, and then returns one of 3 strings.